swift/extractor/main.cpp

```diff
@@ -107,12 +107,9 @@ static void checkWhetherToRunUnderTool(int argc, char* const* argv) {
 // Creates a target file that should store per-invocation info, e.g. compilation args,
 // compilations, diagnostics, etc.
 codeql::TargetFile invocationTargetFile(codeql::SwiftExtractorConfiguration& configuration) {
-  auto hasher = picosha2::hash256_one_by_one();
-  for (auto& option : configuration.frontendOptions) {
-    hasher.process(option.c_str(), option.c_str() + option.size() + 1);
-  }
-  hasher.finish();
-  auto target = "invocation-"s + get_hash_hex_string(hasher);
+  auto timestamp = std::chrono::system_clock::now().time_since_epoch().count();
+  auto filename = std::to_string(timestamp) + '-' + std::to_string(getpid());
```

Comment:
does this work for sorting? I suppose it wouldn't if the time counter gets one more digit during the build, but that's pretty rare for the time since epoch I guess?

It's so unfortunate easy human readable conversion of time points to strings only hit C++ with C++20
https://en.cppreference.com/w/cpp/chrono/system_clock/operator_ltlt

An alternative is this somewhat clunky code
```suggestion
  using Clock = std::chrono::system_clock;
  auto timestamp = Clock::to_time_t(Clock::now());
  std::string filaneme = "YYYY-MM-DDThh:mm:ss";
  std::strftime(filaneme.data(), filaneme.size() + 1, "%FT%T", std::localtime(&now));
  filename += '-';
  filename += std::to_string(getpid());
```
Resolution is at seconds, which however should not be a problem with the PID besides it. Getting decimal seconds in is possible but even more painful :old-man-yells-at-c++:

Feel free to use it or ignore it :slightly_smiling_face:

Reply:
I don't it's worth dealing with `strftime` 😆